File: src/main/java/it/pagopa/ecommerce/controller/PaymentMethodController.java
package it.pagopa.ecommerce.controller;

import java.util.Set;
import java.util.concurrent.CompletionStage;

import it.pagopa.ecommerce.dto.PaymentMethod;
import it.pagopa.ecommerce.services.PaymentMethodService;
import jakarta.inject.Inject;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;

import org.eclipse.microprofile.rest.client.inject.RestClient;
import org.jboss.resteasy.annotations.jaxrs.PathParam;

import io.smallrye.mutiny.Uni;
import org.jboss.logging.Logger;

@Path("/payment-method")
public class PaymentMethodController {

    @Inject
    @RestClient
    PaymentMethodService paymentMethodService;

    private final Logger LOGGER = Logger.getLogger(PaymentMethodController.class);


    @GET
    @Path("/all")
    public Set<PaymentMethod> getAllPaymentMethods(@PathParam String id) {
        LOGGER.info("CALL GET ALL PAYMENT-METHOD");
        return paymentMethodService.getAll();
    }

    @GET
    @Path("/all-async")
    public CompletionStage<PaymentMethod> getAllPaymentMethodsAsync(@PathParam String id) {
        return paymentMethodService.getAllAsync();
    }

    @GET
    @Path("/all-as-uni")
    public Uni<PaymentMethod> getAllPaymentMethodsAsUni(@PathParam String id) {
        return paymentMethodService.getAllAsUni();
    }

    @GET
    @Path("/id/{id}")
    public Set<PaymentMethod> getPaymentMethod(@PathParam String id) {
        return paymentMethodService.getById(id);
    }

    @GET
    @Path("/id-async/{id}")
    public CompletionStage<Set<PaymentMethod>> getPaymentMethodAsync(@PathParam String id) {
        return paymentMethodService.getByIdAsync(id);
    }

    @GET
    @Path("/id-uni/{id}")
    public Uni<Set<PaymentMethod>> getPaymentMethodMutiny(@PathParam String id) {
        return paymentMethodService.getByIdAsUni(id);
    }
}

File: src/main/kotlin/ecommerce/controller/PaymentMethodController.kt
package ecommerce.controller

import io.smallrye.mutiny.Uni
import ecommerce.dto.PaymentMethod
import ecommerce.services.PaymentMethodService
import jakarta.inject.Inject
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.PathParam
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import java.util.concurrent.CompletionStage

@Path("/payment-method")
class PaymentMethodController {

    @Inject
    lateinit var paymentMethodService: PaymentMethodService

    private val logger: Logger = LoggerFactory.getLogger(PaymentMethodController::class.java)


    @GET
    @Path("/all")
    fun getAllPaymentMethods(): Set<PaymentMethod> {
        logger.info("CALL GET ALL PAYMENT-METHOD")
        return paymentMethodService.getAll()
    }

    @GET
    @Path("/all-async")
    fun getAllPaymentMethodsAsync(): CompletionStage<PaymentMethod> {
        return paymentMethodService.getAllAsync()
    }

    @GET
    @Path("/all-as-uni")
    fun getAllPaymentMethodsAsUni(): Uni<PaymentMethod> {
        return paymentMethodService.getAllAsUni()
    }

    @GET
    @Path("/id/{id}")
    fun getPaymentMethod(@PathParam("id") id: String): Set<PaymentMethod> {
        return paymentMethodService.getById(id)
    }

    @GET
    @Path("/id-async/{id}")
    fun getPaymentMethodAsync(@PathParam("id") id: String): CompletionStage<Set<PaymentMethod>> {
        return paymentMethodService.getByIdAsync(id)
    }

    @GET
    @Path("/id-uni/{id}")
    fun getPaymentMethodMutiny(@PathParam("id") id: String): Uni<Set<PaymentMethod>> {
        return paymentMethodService.getByIdAsUni(id)
    }
}
